### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by mchindri**
These fixes are needed for the new iio implementation

### CODE REVIEW

File: tinyiiod.c [hunk 1]
```diff
@@ -32,6 +32,8 @@ struct tinyiiod * tinyiiod_create(struct tinyiiod_ops *ops)
 		return NULL;
 
 	iiod->buf = malloc(IIOD_BUFFER_SIZE);
+	if (!iiod->buf)
+		return NULL;
```
Comment (adrimbarean) [inline tinyiiod.c:37]: Should free(iiod) before returning NULL.

Comment (mchindri) [reply tinyiiod.c:37]: done

**Comment by mchindri** [comment]
V2:
 - free(iiod) on return NULL

**Comment by mchindri** [comment]
v3:
- Remove: commit "tinyiiod.c : Return of read is never check. " and replace with a better implementation in : "tinyiiod.c : Call directly iiod->ops->read instead of creating a wrapper"

**Review by adrimbarean — APPROVED** [review]
Looks good to me.

**Review by dbogdan — APPROVED** [review]

**Pull request merged**